src/main/kotlin/com/coder/gateway/CoderGatewayConnectionProvider.kt

```diff
@@ -2,93 +2,129 @@
+            logger.debug("Launched Coder connection provider", parameters)
+
+            val deploymentURL = parameters[URL]
+                ?: CoderRemoteConnectionHandle.ask("Enter the full URL of your Coder deployment")
+            if (deploymentURL.isNullOrBlank()) {
+                throw IllegalArgumentException("Query parameter \"$URL\" is missing")
             }
-        }
 
-        recentConnectionsService.addRecentConnection(parameters.toRecentWorkspaceConnection())
-        GatewayUI.getInstance().reset()
+            val (client, username) = authenticate(deploymentURL.toURL(), parameters[TOKEN])
+
+            // TODO: If these are missing we could launch the wizard.
+            val name = parameters[WORKSPACE] ?: throw IllegalArgumentException("Query parameter \"$WORKSPACE\" is missing")
+            val agent = parameters[AGENT] ?: throw IllegalArgumentException("Query parameter \"$AGENT\" is missing")
+
+            val workspaces = client.workspaces()
+            val agents = workspaces.flatMap { it.toAgentModels() }
+            val workspace = agents.firstOrNull { it.name == "$name.$agent" }
+                ?: throw IllegalArgumentException("The agent $agent does not exist on the workspace $name or the workspace is off")
+
+            // TODO: Turn on the workspace if it is off then wait for the agent
+            //       to be ready.  Also, distinguish between whether the
+            //       workspace is off or the agent does not exist in the error
+            //       above instead of showing a combined error.
+
+            val cli = CoderCLIManager.ensureCLI(
+                deploymentURL.toURL(),
+                client.buildInfo().version,
+                settings,
+                indicator,
+            )
+
+            indicator.text = "Authenticating Coder CLI..."
+            cli.login(client.token)
+
+            indicator.text = "Configuring Coder CLI..."
+            cli.configSsh(agents)
+
+            // TODO: Ask for these if missing.  Maybe we can reuse the second
+            //  step of the wizard?  Could also be nice if we automatically used
+            //  the last IDE.
+            if (parameters[IDE_PRODUCT_CODE].isNullOrBlank()) {
+                throw IllegalArgumentException("Query parameter \"$IDE_PRODUCT_CODE\" is missing")
+            }
+            if (parameters[IDE_BUILD_NUMBER].isNullOrBlank()) {
+                throw IllegalArgumentException("Query parameter \"$IDE_BUILD_NUMBER\" is missing")
+            }
+            if (parameters[IDE_PATH_ON_HOST].isNullOrBlank() && parameters[IDE_DOWNLOAD_LINK].isNullOrBlank()) {
+                throw IllegalArgumentException("One of \"$IDE_PATH_ON_HOST\" or \"$IDE_DOWNLOAD_LINK\" is required")
+            }
+
+            // TODO: Ask for the project path if missing and validate the path.
+            val folder = parameters[FOLDER] ?: throw IllegalArgumentException("Query parameter \"$FOLDER\" is missing")
```

Comment:
I think it would be a very good idea to have a dialog showing the parameters and allowing the user to inspect them before connecting.